torch/backends/_nnapi/serializer.py

```diff
@@ -469,11 +472,14 @@ def get_tensor_operand_by_jitval_fixed_size(self, jitval):
                 raise Exception("Flexible size is not supported for this operand.")
         return op_id, oper
 
-    def get_tensor_operand_or_constant(self, jitval):
+    def get_tensor_operand_or_constant(self, jitval, dim_value=DimOrder.PRESUMED_CONTIGUOUS):
         operand_id = self.jitval_operand_map.get(jitval)
         if operand_id is None:
             _, value = self.get_constant_value(jitval, "TensorType")
-            operand_id = self.add_tensor_operand_for_weight(value)
+            # assume constants are contiguous
+            if dim_value == DimOrder.CHANNELS_LAST:
+                value.permute(0, 2, 3, 1)
```

Comment:
This is not an in-place op.  I think you need to do `value = value.permute(0, 2, 3, 1)`.

Should probably first update the test so it fails due to this.